### ISSUE TITLE
Unable to exclude files with a wildcard.

### ISSUE THREAD

**Issue by ryan7273** [infection/infection#190]
| Question    | Answer
| ------------| ---------------
| Infection version | dev-master#d9471cdb95474df662bed346b7c5afb8b8d2ec5d
| Test Framework version | PHPUnit 6.5.5
| PHP version | 7.1.11
| Platform    | Ubuntu

- I can add directories with or without wildcards to the `excludes` section of `infection.json.dist` and they are properly ignored.
- I can add file names without wildcards to the `excludes` section and they are properly ignored.
- If I add a file name with a wildcard then it is not excluded. I have 93 mutants not covered by tests which should not be showing up because they are in files which should be ignored.

<details>
 <summary>phpunit.xml</summary>
{
    "timeout": 10,
    "source": {
        "directories": [
            "src/naigc"
        ],
        "excludes": [
            "exception",
            "external",
            "*.interface.php"
        ]
    },
    "tmpDir": ".infection",
    "phpUnit": {
        "customPath": "vendor-bin/phpunit/vendor/phpunit/phpunit/phpunit"
    },
    "logs": {
        "text": "infection-log.txt"
    }
}
</details>

**Comment by borNfreee** [comment]
Thank you for posting this issue. I will take a look

**Comment by borNfreee** [comment]
Did a brief investigation.

For historical reasons, we support glob pattern (see http://php.net/manual/en/function.glob.php) for items in `excludes` section in `infection.json`. But it works with `src` folders prepended.

With such config:

```json
{
"source": {
        "directories": [
            "src",
            "module"
        ],
        "excludes": ["glob/pattern*"]
    }
}
```

infection builds the following patterns:

* `src/glob/pattern/*`
* `module/glob/pattern/*`

I'm not sure this was a good decision though.

After all glob patterns expanded, all items from `excludes` config section are passed to the Symfony Finder's `path()` function. It means that there we can use regular string or regexes https://github.com/symfony/finder/blob/v4.0.4/Finder.php#L235

So, if you want to exclude "*.interface.php", you can try the following:

```json
{
	"source": {
		"directories": ["src/naigc"],
		"excludes": ["/\\.interface\\.php/"]
	}
}
```

It works for me.

**Comment by ryan7273** [comment]
Wow. I never would have figured that out. It works perfectly :+1: 

Might I suggest adding this to the documentation until/unless excludes are handled differently? I doubt I'll be the only one to run into this.

**Comment by borNfreee** [comment]
> Might I suggest adding this to the documentation until/unless excludes are handled differently? I doubt I'll be the only one to run into this.

Done https://github.com/infection/site/commit/d6d35059b1c6b3138a3b88b5ed555b0cc6e3861b

Feel free to suggest rewording if it is not yet clear for the end user

**Comment by ryan7273** [comment]
Looks great.

**Comment by emodric** [comment]
Hi everyone! I feel that docs do not describe well that `excludes` patterns are appended to `source` patterns. Only after I found this issue it became clear that I was writing by `excludes` patterns wrong, with the path relative to working directory, instead relative to `sources` patterns.

**Comment by borNfreee** [comment]
thank you @emodric, I will update the docs
